torch/distributions/constraints.py

```diff
@@ -488,11 +490,21 @@ def check(self, value):
         return torch.isclose(value, value.mT, atol=1e-6).all(-2).all(-1)
 
 
+class _PositiveSemidefinite(_Symmetric):
```

Comment:
That would be great. Thank you for your feedback.

Reply:
@vadimkantorov 
Will it be better to distinguish `_SymmetricPositiveDefinite` and `_PositiveDefinite`? I can change the name of current `_PositiveDefinite` to `_SymmetricPositiveDefinite` and modify its usage inside PyTorch. I assume there are few people utilized the `torch.distributions.constraints` outside the PyTorch and if I make new `_PositiveDefinite` to be compatible with `_SymmetricPositiveDefinite` by just generalize it to non-symmetric matrix, I guess it will not raise issues.

Reply:
I'm not sure if it is straightforward to be able to separate the two since many linalg operations like `cholesky` will assume that the tensor is symmetric.

Reply:
> I commited this proposal.
I mean I made another branch testing it. This PR applies PSD and PD constraints to hide `symmetric`.